test/functional/wallet_send.py

```diff
@@ -0,0 +1,321 @@
+
+        self.log.info("Don't broadcast...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False)
+        assert(res["hex"])
+
+        self.log.info("Return PSBT...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, psbt=True)
+        assert(res["psbt"])
+
+        self.log.info("Create transaction that spends to address, but don't broadcast...")
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, add_to_wallet=False)
+        # conf_target & estimate_mode can be set as argument or option
+        res1 = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical", add_to_wallet=False)
+        res2 = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=1, estimate_mode="economical", add_to_wallet=False)
+        assert_equal(self.nodes[1].decodepsbt(res1["psbt"])["fee"],
+                     self.nodes[1].decodepsbt(res2["psbt"])["fee"])
+        # but not at the same time
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, arg_conf_target=1, arg_estimate_mode="economical",
+                       conf_target=1, estimate_mode="economical", add_to_wallet=False, expect_error=(-8,"Use either conf_target and estimate_mode or the options dictionary to control fee rate"))
+
+        self.log.info("Create PSBT from watch-only wallet w3, sign with w2...")
+        res = self.test_send(from_wallet=w3, to_wallet=w1, amount=1)
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Create PSBT from wallet w4 with watch-only keys, sign with w2...")
+        self.test_send(from_wallet=w4, to_wallet=w1, amount=1, expect_error=(-4, "Insufficient funds"))
+        res = self.test_send(from_wallet=w4, to_wallet=w1, amount=1, include_watching=True, add_to_wallet=False)
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Create OP_RETURN...")
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1)
+        self.test_send(from_wallet=w0, data="Hello World", expect_error=(-8, "Data must be hexadecimal string (not 'Hello World')"))
+        self.test_send(from_wallet=w0, data="23")
+        res = self.test_send(from_wallet=w3, data="23")
+        res = w2.walletprocesspsbt(res["psbt"])
+        assert res["complete"]
+
+        self.log.info("Set fee rate...")
+        res = self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=2, estimate_mode="duff/b", add_to_wallet=False)
+        fee = self.nodes[1].decodepsbt(res["psbt"])["fee"]
+        assert_fee_amount(fee, Decimal(len(res["hex"]) / 2), Decimal("0.00002"))
+        self.test_send(from_wallet=w0, to_wallet=w1, amount=1, conf_target=-1, estimate_mode="duff/b",
+                       expect_error=(-3, "Amount out of range"))
+
+        # TODO: Return hex if fee rate is below -maxmempool
```

Comment:
if there's a missing backport - add a number of backport to comment to know when this TODO will require an action

Reply:
This is not a missing backport, this is the TODO [verbatim](https://github.com/bitcoin/bitcoin/pull/16378/files#diff-45c9fdef949aa07c272e8b6b7fb6a6d7db3ba056ba5746c6feba6ba90a58b85dR263) from [bitcoin#16378](https://github.com/bitcoin/bitcoin/pull/16378)